### PR TITLE
Fixes round restart audio clipping

### PR DESCRIPTION
**Pull request by FairlySadPanda**
## About the PR
Another pass at fixing round restart audio based on using the background audio system client-side.

IDK why I'm addicted to mucking around with the lobby screen

## Technical details
Partial revert of my prior PR (turns out transmitting it out to clients _is_ smarter)

## Media
Can't link sound effects here :(

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- fix: Lobby restart sound effects no longer cut-off.

### CODE REVIEW

**Comment by FairlySadPanda** [comment]
Debate: is swearing ok in the round restart audio, it means that anyone who forks or makes a server has to comment it out if they have no swearing rules (which, yes, isn't a thing at the moment but "this is a Christian Minecraft server" etc etc etc

**Comment by Golinth** [comment]
Seeing as one of the maps literally has a giant "FUCK" in engi using decals, I think swearing should be fine

**Comment by metalgearsloth** [comment]
science_3 is too long, it's gonna overlap 20 seconds of any song.

**Comment by FairlySadPanda** [comment]
Length is a good point. I'll swap it out for the Liltenhead "what the fuck is a throngle" line I thought would work.

Sidebar: using temp VA including dev voices is not cringe, it is part of normal development practice in the industry.

Think the Science copypasta is a good meme to include. Think a total length of 12s or so would work.

I assume the code is fine otherwise?

**Comment by VasilisThePikachu** [comment]
> Also self-inserting yourself into a FOSS game is pretty cringe.

Respectively, I disagree that's it's cringe. Getting a VA is gonna be a challenge if you hire them let alone getting licencing. The voice clip in question is vocoded as well.

But I will agree the voice clip on question is pretty long.

**Comment by FairlySadPanda** [comment]
> > Also self-inserting yourself into a FOSS game is pretty cringe.
> 
> Respectively, I disagree that's it's cringe. Getting a VA is gonna be a challenge if you hire them let alone getting licencing. The voice clip in question is vocoded as well.
> 
> But I will agree the voice clip on question is pretty long.

Well it is a bit cringe, it is cringe as soon as somebody thinks it is! 👀 It also isn't the version that went through the Vocaloid AI conversion work as that had siblance problems.

TODOs for it: Find a final VA, only do the first ten seconds of the pasta

**Comment by FairlySadPanda** [comment]
Length is a good point. I'll swap it out for the Liltenhead "what the fuck is a throngle" line I thought would work.

Sidebar: using temp VA including dev voices is not cringe, it is part of normal development practice in the industry.

Think the Science copypasta is a good meme to include. Think a total length of 12s or so would work.

I assume the code is fine otherwise?

**Comment by FairlySadPanda** [comment]
The problem with the metal pipes sound is that it's an old stock sound from at least before 1995; finding the original license for it is probably impossible.

File: Content.Client/Audio/BackgroundAudioSystem.cs [hunk 1]
```diff
@@ -129,4 +132,18 @@ private void EndLobbyMusic()
...
+        if (!_configManager.GetCVar(CCVars.LobbyMusicEnabled))
+            return;
+
+        var file = _gameTicker.RestartSound;
+        if (string.IsNullOrEmpty(file))
+        {
+            return;
+        }
+
+        _audio.PlayGlobal(file, Filter.Local(), false, _lobbyParams.WithVolume(_lobbyParams.Volume + SharedAudioSystem.GainToVolume(_configManager.GetCVar(CCVars.LobbyMusicVolume))));
```
Comment (AJCM-git) [inline Content.Client/Audio/BackgroundAudioSystem.cs:147]: move lengthy params to local vars for better readability

File: Content.Server/GameTicking/GameTicker.RoundFlow.cs [hunk 2]
```diff
@@ -387,7 +387,7 @@ public void ShowRoundEndScoreboard(string text = "")
             var listOfPlayerInfoFinal = listOfPlayerInfo.OrderBy(pi => pi.PlayerOOCName).ToArray();
 
             RaiseNetworkEvent(new RoundEndMessageEvent(gamemodeTitle, roundEndText, roundDuration, RoundId,
-                listOfPlayerInfoFinal.Length, listOfPlayerInfoFinal, LobbySong));
+                listOfPlayerInfoFinal.Length, listOfPlayerInfoFinal, LobbySong, _audio.GetSound(new SoundCollectionSpecifier("RoundEnd"))));
```
Comment (AJCM-git) [inline Content.Server/GameTicking/GameTicker.RoundFlow.cs:390]: same

**Review by AJCM-git — DISMISSED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**